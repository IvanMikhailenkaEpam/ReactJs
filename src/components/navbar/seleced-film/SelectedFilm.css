.search-aria {
    width: 100%!important;
    background-color: black!important;
    border: 0;
    border-bottom: 2px solid #F85266;
}

.button-pink {
    background-color: #F85266;
    width: 100px;
    font-weight: bold;
    cursor: pointer;
}

.button-black {
    background-color: #3C3B39;
    width: 100px;
    font-weight: bold;
    cursor: pointer;
}

.search-button {
    left: -15px;
}

.search-button > button {
    background-color: #F85266;
    width: 150px;
    font-weight: bold;
    cursor: pointer;
}
img {
    width: 350px;
    height: 450px
}


