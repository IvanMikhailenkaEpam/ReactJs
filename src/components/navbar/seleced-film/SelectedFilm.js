import React from 'react';
import './SelectedFilm.css';
import {Link} from 'react-router-dom';
import * as PropTypes from "prop-types";

const SelectedFilm = props => {
  const {film} = props;
  return (
    <div>
      <div className="row mt-1 w-100">
        <div className="col-4">
          <span className="pink font-weight-bold">netflixroulette</span>
        </div>
        <div className="col-auto ml-auto">
          <Link to="/">
            <button className="btn my-2 my-sm-0 white button-pink" type="button">
              SEARCH
            </button>
          </Link>
        </div>
      </div>
      <div className="row mt-4 mb-4">
        <div className="col-auto">
          <img
            src={film.poster_path}
            alt="text"
          />
        </div>
        <div className="col-7 ml-5">
          <div className="row mt-4">
            <div className="col-12 flex">
              <h1 className="pink font-weight-bold">{film.title}</h1>
              <span className="ml-5 white font-weight-bold">{film.vote_average}</span>
            </div>
            <div className="col-12">
              <span className="white">{film.tagline}</span>
            </div>
            <div className="col-12 mt-3">
              <span className="white mr-5">{film.release_date}</span><span
              className="ml-5 white">{film.runtime}</span>
            </div>
            <div className="col-12 mt-3">
                  <span className="white">
                    {film.overview}
                  </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

SelectedFilm.prototype = {
  film: PropTypes.shape({
    poster_path: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    vote_average: PropTypes.number.isRequired,
    tagline: PropTypes.string,
    release_date: PropTypes.string.isRequired,
    runtime: PropTypes.string.isRequired,
    overview: PropTypes.string.isRequired
  }).isRequired
};

export default SelectedFilm;
