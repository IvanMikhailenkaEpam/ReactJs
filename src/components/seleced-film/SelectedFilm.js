import React, { Component } from 'react';
import './SelectedFilm.css';
import { Link } from 'react-router-dom';

class SelectedFilm extends Component {
  render() {
    return (
      <nav className="navbar navbar-expand-md navbar-dark bg-dark header">
        <div className="container">
          <div className="row mt-1 w-100">
            <div className="col-4">
              <span className="pink font-weight-bold">netflixroulette</span>
            </div>
            <div className="col-auto ml-auto">
              <Link to="/">
                <button className="btn my-2 my-sm-0 white button-pink" type="button">
                  SEARCH
                </button>
              </Link>
            </div>
          </div>
          <div className="row mt-4 mb-4">
            <div className="col-auto">
              <img
                src="https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg"
                alt="text"
              />
            </div>
            <div className="col-7 ml-5">
              <div className="row mt-4">
                <div className="col-12">
                  <h1 className="pink font-weight-bold">Pulp Fiction </h1>
                </div>
                <div className="col-12">
                  <span className="white">Oscar-winning Movies</span>
                </div>
                <div className="col-12 mt-3">
                  <span className="white">1994        154 min</span>
                </div>
                <div className="col-12 mt-3">
                  <span className="white">
                    Jules Winnfield (Samuel L. Jackson) and Vincent Vega (John Travolta) are
                    two hit men who are out to retrieve a suitcase stolen from their employer, mob
                    boss Marsellus Wallace (Ving Rhames). Wallace has also asked Vincent to take his
                    wife Mia (Uma Thurman) out a few days later when Wallace himself will be out of
                    town. Butch Coolidge (Bruce Willis) is an aging boxer who is paid by Wallace to
                    lose his fight. The lives of these seemingly unrelated people are woven together
                    comprising of a series of funny, bizarre and uncalled-for incidents.
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </nav>
    );
  }
}

export default SelectedFilm;
