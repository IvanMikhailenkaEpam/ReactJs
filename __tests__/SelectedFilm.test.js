import React from 'react';
import {shallow} from 'enzyme';
import SelectedFilm from "../src/components/navbar/seleced-film/SelectedFilm";

describe('SelectedFilm', () => {
   const film =
    {
      'id': 1,
      'name': "film 1",
      'genre': 'genre 1',
      'age': 1332,
      'rating': 4.5,
      'img': "https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg",
      'description': "Some info",
      'director': "Tarantion 1",
      'regards': 'Oscar winning move',
      'duration': '122 min',
    };


  it('renders SelectedFilm correctly', () => {
    const component = shallow(<SelectedFilm film={film}/>);

    expect(component).toMatchSnapshot();
  });
});
