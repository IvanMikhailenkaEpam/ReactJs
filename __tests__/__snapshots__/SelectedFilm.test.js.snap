// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedFilm renders SelectedFilm correctly 1`] = `
<div>
  <div
    className="row mt-1 w-100"
  >
    <div
      className="col-4"
    >
      <span
        className="pink font-weight-bold"
      >
        netflixroulette
      </span>
    </div>
    <div
      className="col-auto ml-auto"
    >
      <Link
        to="/"
      >
        <button
          className="btn my-2 my-sm-0 white button-pink"
          type="button"
        >
          SEARCH
        </button>
      </Link>
    </div>
  </div>
  <div
    className="row mt-4 mb-4"
  >
    <div
      className="col-auto"
    >
      <img
        alt="text"
        src="https://m.media-amazon.com/images/M/MV5BNGNhMDIzZTUtNTBlZi00MTRlLWFjM2ItYzViMjE3YzI5MjljXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SY1000_CR0,0,686,1000_AL_.jpg"
      />
    </div>
    <div
      className="col-7 ml-5"
    >
      <div
        className="row mt-4"
      >
        <div
          className="col-12 flex"
        >
          <h1
            className="pink font-weight-bold"
          >
            film 1
          </h1>
          <span
            className="ml-5 white font-weight-bold"
          >
            4.5
          </span>
        </div>
        <div
          className="col-12"
        >
          <span
            className="white"
          >
            Oscar winning move
          </span>
        </div>
        <div
          className="col-12 mt-3"
        >
          <span
            className="white mr-5"
          >
            1332
          </span>
          <span
            className="ml-5 white"
          >
            122 min
          </span>
        </div>
        <div
          className="col-12 mt-3"
        >
          <span
            className="white"
          >
            Some info
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
